src/classes/Class.java

```diff
@@ -18,6 +18,17 @@ public Class(int year, Course course) {
     }
     // </editor-fold>
 
+    // <editor-fold desc="ToString">
+    @Override
+    public String toString() {
+        return "Turma: " +
+                "Ano: " + year +
+                ", Curso: " + course +
+                ", Alunos: " + students;
+    }
```

Comment:
Testando esse código eu tô percebendo que tá acontecendo algo que acho que não é desejável.

Do jeito que está, ao printar uma Class, quando ele na linha 26 `", Curso: " + course +`, ele transforma o course em string também, chamando o toString() da classe Course.

Que por usa vez chama o toString() da classe Teacher.

Depois, ao printar a lista inteira de students, ele chama o toString de cada Student dentro da lista. E cada Student chama o toString do Course de novo, criando uma poluição enorme no console.

Você pode testar isso adicionando a linha ao final da classe Main:
`System.out.println(class2024);`

Ao rodar, gera esse print no final:
`Turma: Ano: 2024, Curso: Curso: Nome: BackEnd-Java, Professor: Professor: Nome: André, Idade: 30, Anos de trabalho: 0, Alunos: [Estudante: Nome: Gabriel, Cursos: [Curso: Nome: BackEnd-Java, Professor: Professor: Nome: André, Idade: 30, Anos de trabalho: 0], Estudante: Nome: Samuel, Cursos: [Curso: Nome: FrontEnd-Angular, Professor: Professor: Nome: André, Idade: 30, Anos de trabalho: 0]]`

A minha sugestão:
- Primeiro, em todo toString que for printar student, teacher ou course, usar o getName() para printar somente o nome do objeto, e não chamar o toString inteiro.
- Segundo, eu acho que não é ideal, ao printar a classe, também printar a lista inteira de alunos ingressados. Acho que seria mais interessante apenas printar o número de alunos.

Reply:
Bem lembrado, no primeiro pull request que eu fiz eu usei o   ", Curso: " + course.getName()
Mas como houve uma alteração no codigo referente a EX:4 que foi criado novas classes o meu pull request ficou quebrado, entao tive que fazer tudo de novo. :)
Vou fazer a alteração aqui e envio novamente o pull request